src/linechart/LineChart.jsx

```diff
@@ -61,86 +47,71 @@ module.exports = React.createClass({
-      data = [data];
+    if (!Array.isArray(props.data)) {
+      props.data = [props.data];
     }
 
-    var flattenedData = utils.flattenData(data, props.xAccessor, props.yAccessor);
+    // Returns an object of flattened allValues, xValues, and yValues
+    var flattenedData = utils.flattenData(props.data, props.xAccessor, props.yAccessor);
 
     var allValues = flattenedData.allValues,
         xValues = flattenedData.xValues,
         yValues = flattenedData.yValues;
-
-    var scales = utils.calculateScales(innerWidth, innerHeight, xValues, yValues);
-
-    var trans = `translate(${ props.margins.left },${ props.margins.top })`;
-
-    var dataSeriesArray = data.map( (series, idx) => {
-      return (
-          <DataSeries
-            key={idx}
-            structure={structure}
-            xScale={scales.xScale}
-            yScale={scales.yScale}
-            seriesName={series.name}
-            data={series.values}
-            width={innerWidth}
-            height={innerHeight}
-            fill={props.colors(props.colorAccessor(series, idx))}
-            circleRadius={props.circleRadius}
-            xAccessor={props.xAccessor}
-            yAccessor={props.yAccessor}
-            interpolationType={interpolationType}
-            displayDataPoints={props.displayDataPoints}
-          />
-      );
-    });
+    var scales = this._calculateScales(innerWidth, innerHeight, xValues, yValues);
+    var trans = "translate(" + (props.yAxisOffset < 0 ? props.margins.left + Math.abs(props.yAxisOffset) : props.margins.left) + "," + props.margins.top + ")";
 
     return (
       <Chart
         viewBox={props.viewBox}
         legend={props.legend}
-        data={data}
+        data={props.data}
         margins={props.margins}
         colors={props.colors}
-        colorAccessor={props.colorAccessor}
+        colorAccessor={(d, i) => d}
         width={props.width}
         height={props.height}
-        title={props.title}
-      >
-        <g transform={trans} className={props.className}>
-          {dataSeriesArray}
-          {props.hoverAnimation ? <Voronoi
-            structure={structure}
-            data={allValues}
+        title={props.title}>
+        <g transform={trans} className='rd3-linechart'>
+          <DataSeries
             xScale={scales.xScale}
             yScale={scales.yScale}
+            xAccessor={props.xAccessor}
+            yAccessor={props.yAccessor}
+            hoverAnimation={props.hoverAnimation}
+            circleRadius={props.circleRadius}
+            data={props.data}
+            value={allValues}
+            colors={props.colors}
+            colorAccessor={(d, i) => d}
             width={innerWidth}
             height={innerHeight}
-          /> : <g/> }
+          />
           <XAxis
-            xAxisClassName='rd3-linechart-xaxis'
-            tickFormatting={props.xAxisFormatter}
+            xAxisClassName="rd3-linechart-xaxis"
+            strokeWidth="1"
```

Comment:
We should never hard code values like this. Make it a property so users can override it if they want.